diff --git a/libavfilter/Makefile b/libavfilter/Makefile
index a99362b3ee..38aea815e1 100644
--- a/libavfilter/Makefile
+++ b/libavfilter/Makefile
@@ -431,6 +431,7 @@ OBJS-$(CONFIG_YADIF_CUDA_FILTER)             += vf_yadif_cuda.o vf_yadif_cuda.pt
 OBJS-$(CONFIG_ZMQ_FILTER)                    += f_zmq.o
 OBJS-$(CONFIG_ZOOMPAN_FILTER)                += vf_zoompan.o
 OBJS-$(CONFIG_ZSCALE_FILTER)                 += vf_zscale.o
+OBJS-$(CONFIG_GENERICSHADER_FILTER)          += vf_genericshader.o
 
 OBJS-$(CONFIG_ALLRGB_FILTER)                 += vsrc_testsrc.o
 OBJS-$(CONFIG_ALLYUV_FILTER)                 += vsrc_testsrc.o
diff --git a/libavfilter/allfilters.c b/libavfilter/allfilters.c
index 858ed1cf78..fbb0a166ed 100644
--- a/libavfilter/allfilters.c
+++ b/libavfilter/allfilters.c
@@ -465,6 +465,7 @@ extern  AVFilter ff_asink_abuffer;
 extern  AVFilter ff_vsink_buffer;
 extern AVFilter ff_af_afifo;
 extern AVFilter ff_vf_fifo;
+extern AVFilter ff_vf_genericshader;
 
 #include "libavfilter/filter_list.c"
 
